File: leetcode24.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
        ListNode* first;
        ListNode* second;
        if (head == NULL || head->next == NULL)
        {
        	return head;
        }
        first = head;
        second = head->next;
        while(second != NULL && first != NULL)
        {
        	int tmp = first->val;
        	first->val = second->val;
        	second->val = tmp; 
        	if(first->next!=NULL)
        		first = first->next->next;
        	if(second->next!=NULL)
        		second = second->next->next;
        }
        return head;
    }
};

File: leetcode24.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode swapPairs(ListNode head) {
        if(head == null || head.next == null){
            return head;
        }
        ListNode vHead = new ListNode(0);
        vHead.next = head;
        ListNode pre = vHead;
        while(pre.next != null && pre.next.next != null){
            ListNode cur = pre.next;
            ListNode tmp = pre.next.next;
            cur.next = tmp.next;
            pre.next = tmp;
            pre.next.next = cur;
            pre = pre.next.next;
        }
        return vHead.next;
    }
}
